manifests/init.pp

```diff
@@ -28,53 +28,98 @@
 # [*root_store*]
 # The store location for the given certifcation store. Either LocalMachine or CurrentUser
 #
+# [*scripts_dir*]
+# The directory where the scripts to verify and install the certificates will be stored.
+# By default is C:\temp
+#
+# [*is_exportable*]
+# Flag to set the key as exportable. true == exportable; false == not exportable.
+# By default is set to true.
 # === Examples
 #
 # To install a certificate in the My directory of the LocalMachine root store:
 #
 #  sslcertificate { "Install-PFX-Certificate" :
 #    name       => 'mycert.pfx',
 #    password   => 'password123',
-#    location   => 'C:\',
+#    location   => 'C:',
 #    thumbprint => '07E5C1AF7F5223CB975CC29B5455642F5570798B'
 #  }
 #
-# To install a certifcate in an alterntative direcotory:
+# To install a certifcate in an alternative directory:
 #
 #  sslcertificate { "Install-Intermediate-Certificate" :
 #    name       => 'go_daddy_intermediate.p7b',
-#    location   => 'C:\',
+#    location   => 'C:',
 #    store_dir  => 'CA',
 #    root_store => 'LocalMachine',
 #    thumbprint => '07E5C1AF7F5223CB975CC29B5455642F5570798B'
 #  }
 #
-define sslcertificate($password, $location, $thumbprint, $root_store = 'LocalMachine', $store_dir = 'My') {
-  validate_re($name, '^(.)+$',"Must pass name to ${module_name}[${title}]")
-  validate_re($location, '^(.)+$',"Must pass location to ${module_name}[${title}]")
+# To install a certificate in the My directory of the LocalMachine root store
+# using a different directory to store the scripts:
+#
+#  sslcertificate { "Install-PFX-Certificate" :
+#    name         => 'mycert.pfx',
+#    password     => 'password123',
+#    location     => 'C:',
+#    thumbprint   => '07E5C1AF7F5223CB975CC29B5455642F5570798B',
+#    scripts_dir  => 'C:\scripts_dir'
+#  }
+#
+# To install a certificate in the My directory of the LocalMachine root store
+# and set the key as not exportable:
+#
+#  sslcertificate { "Install-PFX-Certificate" :
+#    name           => 'mycert.pfx',
+#    password       => 'password123',
+#    location       => 'C:',
+#    thumbprint     => '07E5C1AF7F5223CB975CC29B5455642F5570798B',
+#    is_exportable  => false
+#  }
+#
+define sslcertificate (
+  $password,
+  $location,
+  $thumbprint,
+  $root_store    = 'LocalMachine',
+  $store_dir     = 'My',
+  $scripts_dir   = 'C:\temp',
+  $is_exportable = true) {
```

Comment:
I'd prefer this to be just `exportable` instead.

Is 'My' a good default for store_dir?

This should probably be converted to use Puppet 4 datatypes, so that you can get rid of the validate calls below